pkg/operator/reporters/incoming/incoming_reporter.go

```diff
@@ -0,0 +1,128 @@
+package incoming
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/eparis/bugzilla"
+
+	"github.com/mfojtik/bugzilla-operator/pkg/cache"
+	"github.com/mfojtik/bugzilla-operator/pkg/operator/config"
+	"github.com/mfojtik/bugzilla-operator/pkg/operator/controller"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+type IncomingReporter struct {
+	controller.ControllerContext
+	config     config.OperatorConfig
+	components []string
+}
+
+func (c *IncomingReporter) sync(ctx context.Context, syncContext factory.SyncContext) error {
+	client := c.NewBugzillaClient(ctx)
+	slackClient := c.SlackClient(ctx)
+
+	channelReport, assigneeReport, err := Report(ctx, client, syncContext.Recorder(), &c.config, c.components)
+	if err != nil {
+		return err
+	}
+	if len(assigneeReport) == 0 {
+		return nil
+	}
+
+	for assignee, bugs := range assigneeReport {
+		message := fmt.Sprintf("%s\n\n> Please set severity/priority on the bug(s) above and assign to a team member.\n", strings.Join(bugs.reports, "\n"))
+		if err := slackClient.MessageEmail(assignee, message); err != nil {
+			syncContext.Recorder().Warningf("DeliveryFailed", "Failed to deliver:\n\n%s\n\n to %q: %v", message, assignee, err)
+			continue
+		}
+		for _, id := range bugs.bugIDs {
+			if err := c.markAsReported(client, id); err != nil {
+				syncContext.Recorder().Warningf("MarkNotifiedFailed", "Failed to mark bug #%d with AssigneeNotified: %v", id, err)
+			}
+		}
+	}
+
+	if err := slackClient.MessageChannel(channelReport); err != nil {
+		syncContext.Recorder().Warningf("DeliveryFailed", "Failed to deliver new bugs: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func NewIncomingReporter(ctx controller.ControllerContext, components []string, schedule []string, operatorConfig config.OperatorConfig, recorder events.Recorder) factory.Controller {
+	c := &IncomingReporter{
+		ControllerContext: ctx,
+		config:            operatorConfig,
+		components:        components,
```

Comment:
nit: take components from the config directly

Reply:
we use the same logic in other reporters, I can do that but I would like to be consistent, so change others as well...

follow up is fine?

Reply:
actually, do it now, deal with others as follow up